### PR TITLE
Fix type error using stream_set_blocking.

### PR DESCRIPTION
**Pull request by yukitaka**
Change the second argument of stream_set_blocking to int.

### CODE REVIEW

**Comment by yukitaka** [comment]
All right.
I agreed to Contributor License.

File: hphp/system/php/redis/Redis.php [hunk 1]
```diff
@@ -1650,7 +1650,7 @@ protected function doConnect($host,
         E_WARNING);
       return false;
     }
-    stream_set_blocking($this->connection, true);
+    stream_set_blocking($this->connection, 1);
```
Comment (steelbrain) [inline hphp/system/php/redis/Redis.php]: http://php.net/manual/en/function.stream-set-blocking.php <-- docs say it should be bool

**Comment by yukitaka** [comment]
hhvm/hphp/runtime/ext/stream/ext_stream.php <-- Implementation is int

It seems to have different defined by the doc language.

http://php.net/manual/ja/function.stream-set-blocking.php
http://php.net/manual/fr/function.stream-set-blocking.php

php7.0.1
ext/standard/streamsfuncs.c <-- proto comment is int

It is preferable to fix either ?

**Comment by SiebelsTim** [comment]
@yukitaka 
6 days ago: https://github.com/php/php-src/commit/4171a8f6c631e3c34d1d070d553e7611b5474263

---

`int` doesn't make much sense either. So I would suggest sticking with `bool`.

**Comment by yukitaka** [comment]
I prefer bool too.
It looks good to add a bool for backward compatibility .

**Comment by JohanTan** [comment]
Any updates on when this is merging? Redis extension is currently broken on HHVM 3.12 now.

**Comment by SiebelsTim** [comment]
This PR is abandoned. How is redis broken by this?

**Comment by JohanTan** [comment]
Got the following fatal error once `hhvm.php7.all = 1` is enabled:
```Fatal error: Uncaught TypeError: Argument 2 passed to stream_set_blocking() must be an instance of int, bool given```

**Pull request closed without merging**